diff --git a/hadoop-hdfs-project/hadoop-hdfs/src/main/java/org/apache/hadoop/hdfs/server/namenode/NameNodeHttpServer.java b/hadoop-hdfs-project/hadoop-hdfs/src/main/java/org/apache/hadoop/hdfs/server/namenode/NameNodeHttpServer.java
index c05398a31ce..186cb027e2e 100644
--- a/hadoop-hdfs-project/hadoop-hdfs/src/main/java/org/apache/hadoop/hdfs/server/namenode/NameNodeHttpServer.java
+++ b/hadoop-hdfs-project/hadoop-hdfs/src/main/java/org/apache/hadoop/hdfs/server/namenode/NameNodeHttpServer.java
@@ -105,7 +105,7 @@ public static void initWebHdfs(Configuration conf, String hostname,
     final Map<String, String> params = new HashMap<>();
     params.put(ResourceConfig.FEATURE_MATCH_MATRIX_PARAMS, "true");
     httpServer2.addJerseyResourcePackage(
-        jerseyResourcePackage + ";" + Param.class.getPackage().getName(),
+        jerseyResourcePackage + ";" + "org.apache.hadoop.hdfs.web.resources",
         pathSpec, params);
   }
 
@@ -162,7 +162,7 @@ void start() throws IOException {
     String httpKeytab = conf.get(DFSUtil.getSpnegoKeytabKey(conf,
         DFSConfigKeys.DFS_NAMENODE_KEYTAB_FILE_KEY));
     initWebHdfs(conf, bindAddress.getHostName(), httpKeytab, httpServer,
-        NamenodeWebHdfsMethods.class.getPackage().getName());
+        "org.apache.hadoop.hdfs.server.namenode.web.resources");
 
     httpServer.setAttribute(NAMENODE_ATTRIBUTE_KEY, nn);
     httpServer.setAttribute(JspHelper.CURRENT_CONF, conf);
